src/Obsidian/FileParser.ts

```diff
@@ -149,16 +149,16 @@ export class FileParser {
                 }
             } else {
                 // Treat tasks without the global filter as list items
-                this.createListItem(listItem, line, lineNumber);
+                this.createListItem(listItem, line, lineNumber, taskLocation);
             }
         } catch (e) {
             this.errorReporter(e, this.filePath, listItem, line);
         }
         return sectionIndex;
     }
 
-    private createListItem(listItem: ListItemCache, line: string, lineNumber: number) {
+    private createListItem(listItem: ListItemCache, line: string, lineNumber: number, taskLocation: TaskLocation) {
```

Comment:
At some point we can remove the lineNumber parameter here, as taskLocation now also supplies the lineNumber.
I put an assert in, and confirmed that across all our tests, the two values always agree.